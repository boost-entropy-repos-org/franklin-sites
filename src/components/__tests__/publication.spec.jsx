import React from 'react';
import { render, fireEvent, waitForElement } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import Publication from '../publication';
import publicationData from '../__mocks__/publications';

let rendered;

describe('Publication component', () => {
  beforeEach(() => {
    const {
      title,
      authors,
      abstract,
      infoData,
      journalInfo,
      pubmedId,
      statistics,
    } = publicationData;

    rendered = render(
      <Router>
        <Publication
          title={title}
          authors={authors}
          abstract={abstract}
          infoData={infoData}
          journalInfo={journalInfo}
          pubmedId={pubmedId}
          statistics={statistics}
        />
      </Router>
    );
  });

  test('should render', () => {
    const { asFragment } = rendered;
    expect(asFragment()).toMatchSnapshot();
  });

  test('should expand authors', async () => {
    const { queryByText, getByText } = rendered;
    expect(queryByText('Darwin K.')).toBeNull();
    fireEvent.click(getByText('[...]'));
    const author = await waitForElement(() => queryByText('Darwin K.'));
    expect(author).toBeTruthy();
  });

  test('should expand abstract', async () => {
    const { queryByText, getByText } = rendered;
    expect(queryByText(/Captain Fitzroy/)).toBeNull();
    fireEvent.click(getByText('Show abstract'));
    const abstract = await waitForElement(() => queryByText(/Captain Fitzroy/));
    expect(abstract).toBeTruthy();
  });
});

export default publicationData;
