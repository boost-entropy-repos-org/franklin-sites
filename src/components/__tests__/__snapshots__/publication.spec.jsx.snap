// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Publication component should render 1`] = `
<DocumentFragment>
  <section
    class="publication"
  >
    <h4>
      The Zoology of the Voyage of H.M.S. Beagle.
    </h4>
    <section
      class="publication__authors"
    >
      <a
        href="/"
      >
        Darwin C.
      </a>
      , 
      <a
        href="/"
      >
        Darwin B.
      </a>
      , 
      <a
        href="/"
      >
        Darwin A.
      </a>
      , 
      <a
        href="/"
      >
        Darwin D.
      </a>
      , 
      <a
        href="/"
      >
        Darwin E.
      </a>
      , 
      <a
        href="/"
      >
        Darwin F.
      </a>
      , 
      <a
        href="/"
      >
        Darwin G.
      </a>
      , 
      <a
        href="/"
      >
        Darwin H.
      </a>
      , 
      <a
        href="/"
      >
        Darwin I.
      </a>
      <button
        class="button tertiary"
        type="button"
      >
        [...]
      </button>
      , 
      <a
        href="/"
      >
        Darwin X.
      </a>
    </section>
    <section
      class="publication__columns"
    >
      <section
        class="publication__columns__main"
      >
        <section>
          <button
            class="button tertiary"
            type="button"
          >
            Show abstract
          </button>
        </section>
        <div
          class="info-list"
        >
          <div
            class="info-list__item"
          >
            <div
              class="info-list__item__title"
            >
              <h5
                class="bold"
              >
                Tortoise
              </h5>
            </div>
            <div
              class="info-list__item__content"
            >
              <em>
                Reptile species of the family Testudinidae of the order Testudines (the turtles).
              </em>
            </div>
          </div>
          <div
            class="info-list__item"
          >
            <div
              class="info-list__item__title"
            >
              <h5
                class="bold"
              >
                Beetles
              </h5>
            </div>
            <div
              class="info-list__item__content"
            >
              Charles really liked them.
            </div>
          </div>
        </div>
      </section>
      <section
        class="publication__columns__side"
      >
        <ul
          class="no-bullet"
        >
          <li>
            <a
              href="//dx.doi.org/10.1038/331525a0"
            >
              Nature 331:525-527(1988)
            </a>
          </li>
          <li>
            <a
              href="//www.ncbi.nlm.nih.gov/pubmed/123456"
              rel="noopener noreferrer"
              target="_blank"
            >
              PubMed
            </a>
          </li>
          <li>
            <a
              href="//europepmc.org/article/MED/123456"
              rel="noopener noreferrer"
              target="_blank"
            >
              Europe PMC
            </a>
          </li>
          <li>
            <section
              class="publication__statistics"
            >
              <div>
                Cited in other entries:
              </div>
              <span
                class="bubble--medium colour-sea-blue"
              >
                99+
              </span>
            </section>
          </li>
        </ul>
      </section>
    </section>
  </section>
</DocumentFragment>
`;
